docs/fuse-box/tutorials/api-trade-quotes.mdx

```diff
@@ -0,0 +1,338 @@
+Axios is a promise-based HTTP client for the browser and Node.js. Install Axios to handle the API requests:
+```bash
+npm install axios
+```
+
+## Step 3: Fetch the API response.
+
+In this step, we will test calling the API and parsing the response to the web browser console. Add the following code to the `index.tsx` file inside the `Home() {}` Component.
+
+```javascript
+const apiUrl = `https://api.fuse.io/api/v1/trade/quote?apiKey=YOUR_API_KEY&sellToken=FUSE&buyToken=0x34Ef2Cc892a88415e9f02b91BfA9c91fC0bE6bD4&sellAmount=1000000000000000000`
+
+async function fetchData() {
+    try {
+      const response = await fetch(apiUrl);
+      if (!response.ok) {
+        throw new Error('Network response was not ok');
+      }
+      const data = await response.json();
+      console.log(data);
+      } catch (error) {
+      console.error('There was a problem with the fetch operation:', error);
+    }
+  }
+
+fetchData();
+```
+
+Save the file.
+
+Run the application using the command:
+
+```bash
+npm run dev
+```
+Open your browser, go to `http://localhost:3000`, and open the developer console. You will find the results logged to the console.
+
+## Step 4: Implementing the Application
+### State Management with React Hooks
+We use the `useState` hook to manage the form inputs and API response data. This allows us to keep track of the user inputs and the API results.
+
+```javascript
+const [inputs, setInputs] = useState({
+  sellToken: 'FUSE',
+  buyToken: '',
+  sellAmount: '1000000000000000000'
+});
+const [result, setResult] = useState(null);
+```
+
+- `inputs` stores the values of the form fields.
+- `result` will hold the data fetched from the API.
+
+### Handling Form Input Changes
+The `handleInputChange` function updates the `input` state whenever a user types into the form fields. This function is triggered on every keystroke, ensuring that the state is always up-to-date with what the user has entered.
+
+```javascript
+const handleInputChange = (event) => {
+  const { name, value } = event.target;
+  setInputs(prevState => ({
+    ...prevState,
+    [name]: value
+  }));
+};
+```
+
+We use a function to ensure the new state is based on the correct previous state, which is important in React to prevent state update issues during rapid state changes.
+
+### Submitting the Form
+The `handleSubmit` function is called when the user submits the form. It prevents the default form submission event, constructs the API request, and handles the API response.
+
+```javascript
+const handleSubmit = async (event) => {
+  event.preventDefault();
+  const { sellToken, buyToken, sellAmount } = inputs;
+  const url = `https://api.fuse.io/api/v1/trade/quote`;
+  const params = { apiKey: 'your_api_key', sellToken, buyToken, sellAmount };
+
+  try {
+    const response = await axios.get(url, { params });
+    setResult(response.data);
+  } catch (error) {
+    console.error('Error fetching trade quote:', error);
+    setResult({ error: error.message });
```

Comment:
And here 
![Screenshot 2024-06-02 at 16 37 03](https://github.com/fuseio/fuse-docs/assets/68244817/4d2c686f-577c-4247-a486-12aba9a42727)

Reply:
Checked this function in full code and found lack of its functionality.
[This 'try'](https://github.com/fuseio/examples/blob/69081a62a7d88407faa49635bd95c1f60ca612e1/trade-api/pages/index.tsx#L55) is completely missing for example.